### ISSUE TITLE
create_datasets_from_start.sh throws error

### ISSUE THREAD

**Issue by ashaw596** [NVIDIA/DeepLearningExamples#168]
Currently create_datasets_from_start throws the error.
```
Traceback (most recent call last):
  File "/workspace/bert/data/bertPrep.py", line 345, in <module>
    main(args)
  File "/workspace/bert/data/bertPrep.py", line 188, in main
    create_record_worker(args.output_file_prefix + '_training', i)
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
```
Since output_file_prefix is not set. Looking at the pretraining file I can't seem to figure out how the data is suppose to be setup as well since it appears to look for
books_wiki_en_corpus and merged_wiki+books_phase2. merged_wiki+books_phase2  Doesn't appear to be referenced anywhere else however.

**Comment by ashaw596** [comment]
It also throws an error
```
  File "/workspace/bert/data/bertPrep.py", line 345, in <module>
    main(args)
  File "/workspace/bert/data/bertPrep.py", line 188, in main
    create_record_worker(args.output_file_prefix + '_training', i)
  File "/workspace/bert/data/bertPrep.py", line 171, in create_record_worker
    bert_preprocessing_command += ' --vocab_file=' + args.vocab_file
```

**Comment by sharathts** [comment]
Thanks for pointing out the issue. A fix has been issued https://github.com/NVIDIA/DeepLearningExamples/pull/171 and it should be merged soon. For now, you should be able to pull the required changes for it to work as expected.

**Comment by magic282** [comment]
It reports other errors:
```
Traceback (most recent call last):
  File "/workspace/bert/data/bertPrep.py", line 351, in <module>
    main(args)
  File "/workspace/bert/data/bertPrep.py", line 49, in main
    downloader.download()
  File "/workspace/bert/data/Downloader.py", line 19, in download
    self.download_bookscorpus()
  File "/workspace/bert/data/Downloader.py", line 55, in download_bookscorpus
    downloader.download()
  File "/workspace/bert/data/BooksDownloader.py", line 16, in download
    bookscorpus_download_process.communicate()
AttributeError: 'CompletedProcess' object has no attribute 'communicate'
[nltk_data] Downloading package punkt to /root/nltk_data...
[nltk_data]   Package punkt is already up-to-date!
Working Directory: /workspace/bert/data
Action: download
Dataset Name: wikicorpus_en

Directory Structure:
{ 'download': '/workspace/bert/data/download',
  'extracted': '/workspace/bert/data/extracted',
  'formatted': '/workspace/bert/data/formatted_one_article_per_line',
  'hdf5': '/workspace/bert/data/hdf5_lower_case_1_seq_len_512_max_pred_20_masked_lm_prob_0.15_random_seed_12345_dupe_factor_5',
  'sharded': '/workspace/bert/data/sharded_training_shards_256_test_shards_256_fraction_0.2',
  'tfrecord': '/workspace/bert/data/tfrecord_lower_case_1_seq_len_512_max_pred_20_masked_lm_prob_0.15_random_seed_12345_dupe_factor_5'}

Downloading: https://dumps.wikimedia.org/enwiki/latest/enwiki-latest-pages-articles.xml.bz2
** Download file already exists, skipping download
Unzipping: wikicorpus_en.xml.bz2
Traceback (most recent call last):
  File "/workspace/bert/data/bertPrep.py", line 351, in <module>
    main(args)
  File "/workspace/bert/data/bertPrep.py", line 49, in main
    downloader.download()
  File "/workspace/bert/data/Downloader.py", line 22, in download
    self.download_wikicorpus('en')
  File "/workspace/bert/data/Downloader.py", line 60, in download_wikicorpus
    downloader.download()
  File "/workspace/bert/data/WikiDownloader.py", line 47, in download
    open(self.save_path + '/wikicorpus_' + self.language + '.xml', mode='wb', buffering=131072).write(zip.read())
  File "/opt/conda/lib/python3.6/bz2.py", line 182, in read
    return self._buffer.read(size)
MemoryError
[nltk_data] Downloading package punkt to /root/nltk_data...
[nltk_data]   Package punkt is already up-to-date!
Working Directory: /workspace/bert/data
Action: download
```
And our server has 110GB memory. Is it enough?

**Comment by nvcforster** [comment]
We have created a new [PR](https://github.com/NVIDIA/DeepLearningExamples/pull/174) that is currently going through final review. You can try the code from the PR before it gets merged (likely Monday) if you would like. The memory requirement has been reduced for decompressing the large wiki dataset. Please let us know if you encounter any more issues.

**Comment by magic282** [comment]
I just pull the master since I found this PR has been merged. However, it still reports the memory error:
```
Downloading: https://dumps.wikimedia.org/enwiki/latest/enwiki-latest-pages-articles.xml.bz2
** Download file already exists, skipping download
Unzipping: wikicorpus_en.xml.bz2
Traceback (most recent call last):
  File "/workspace/bert/data/bertPrep.py", line 351, in <module>
    main(args)
  File "/workspace/bert/data/bertPrep.py", line 49, in main
    downloader.download()
  File "/workspace/bert/data/Downloader.py", line 21, in download
    self.download_wikicorpus('en')
  File "/workspace/bert/data/Downloader.py", line 59, in download_wikicorpus
    downloader.download()
  File "/workspace/bert/data/WikiDownloader.py", line 47, in download
    open(self.save_path + '/wikicorpus_' + self.language + '.xml', mode='wb', buffering=131072).write(zip.read())
  File "/opt/conda/lib/python3.6/bz2.py", line 182, in read
    return self._buffer.read(size)
MemoryError
```

**Comment by magic282** [comment]
I just unzip the wiki file outside as a workaround. It seems that unzipping it with python is not a good option. Maybe calling external system unzip command is better.

**Comment by ashaw596** [comment]
Created a seperated issue #181 for a separate wikipedia path issue that doesn't throw an error.

**Comment by swethmandava** [comment]
Fixed in #182
